### DISCUSSION TITLE
Why does the image extension not resize external images?

### DISCUSSION THREAD

**Discussion by FlTr** [datenstrom/community#736] [Ask a question] [answered]
The image plugin allows to apply custom sizes for each picture. But this works for self hosted images only. I want to do something like:

    [image https://example.com/link/to/a/picture.jpg Example - 50%]

This doesn't work as expected. Width and height are set to zero and therefore ignored during html generation. [Link to code line in plugin](https://github.com/datenstrom/yellow-extensions/blob/master/source/image/image.php#L45).

Of course, creating a thumbnail for external images does not make sense. But is this intended behaviour?

**Comment by markseuffert** [comment, marked as answer]
In short don't do it this way. Resizing external images is not a supported feature.

The reason takes a bit longer to explain. The [image extension](https://github.com/annaesvensson/yellow-image) was made to improve image handling in Markdown. A typical uses case is creating downscaled image thumbnails on your web server for faster page loading time. Another use case is creating cropped image thumbnails on your web server for different aspect ratios, e.g. landscape, portrait, square. For this to work in practice, all images should be stored in your `media/images` folder.

On the technical side, the HTML [img element expects width and height in pixel](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Img), no other unit is supported and they are [not intended to be used to stretch the image](https://www.w3.org/TR/2021/SPSD-html52-20210128/semantics-embedded-content.html#sec-dimension-attributes). In theory you could use CSS for image transformation. Lots of possibilities for the adventurous web developer, but you're probably getting better results by using images with specific dimensions. Some [online image services](https://unsplash.com) let you specify image dimensions in the URL. Here's an example:

![Screenshot external image resize](https://user-images.githubusercontent.com/2429165/161372204-125dbcc5-3200-4e3a-82b3-cca341b9958f.jpg)

```
<img src="https://images.unsplash.com/photo-1627156863760-f49b81d8ab77?fm=jpg&w=512">
<img src="https://images.unsplash.com/photo-1627156863760-f49b81d8ab77?fm=jpg&w=128">
<img src="https://images.unsplash.com/photo-1627156863760-f49b81d8ab77?fm=jpg&w=64">
```

**Reply by FlTr** [reply]
Thanks for the detailed answer. Then I'll have to find a better way for my usecase.
